tests/combine_span_test.cpp

```diff
@@ -0,0 +1,280 @@
+template <typename T>
+class CombineSpansSymmetric : public ::testing::Test {};
+
+using BothDirections =
+    ::testing::Types<span_test::Forwards, span_test::Backwards>;
+TYPED_TEST_SUITE(CombineSpansSymmetric, BothDirections);
+
+TYPED_TEST(CombineSpansSymmetric, MetropolisAcceptsBarkerRejects) {
+  // cases where the two update rules differ
+  constexpr auto direction = TypeParam::value;
+
+  span_test::mock_rng rng{0.5};
+  nuts::SpanW<double> span_from = span_test::dummy_span(1.2);
+  nuts::SpanW<double> span_to = span_test::dummy_span(1.0);
+  double new_logp = nuts::log_sum_exp(span_from.logp_, span_to.logp_);
+
+  // log(0.5) = -0.3
+  // metropolis: 1.2 - 1.0 = 0.2 > -0.3, accepts
+  {
+    nuts::SpanW<double> span_old = span_from;
+    nuts::SpanW<double> span_new = span_to;
+    nuts::SpanW<double> combined =
+        nuts::combine<nuts::Update::Metropolis, direction>(
+            rng, std::move(span_old), std::move(span_new));
+
+    EXPECT_SELECTED_POINT_EQUAL(combined, span_to, new_logp);
+  }
+
+  // barker: 1.2 - lsexp(1.2, 1.0) = 1.2 - 1.78 = -0.58 < -0.3, rejects
+  {
+    nuts::SpanW<double> span_old = span_from;
+    nuts::SpanW<double> span_new = span_to;
+    nuts::SpanW<double> combined =
+        nuts::combine<nuts::Update::Barker, direction>(rng, std::move(span_old),
+                                                       std::move(span_new));
+
+    EXPECT_SELECTED_POINT_EQUAL(combined, span_from, new_logp);
+  }
+}
+
+// metropolis is always strightly higher than barker, so no case where barker
+// accepts and metropolis rejects for a given random uniform draw
+
+TYPED_TEST(CombineSpansSymmetric, BarkerAndMetropolisBothReject) {
+  // cases where the two update rules differ
+  constexpr auto direction = TypeParam::value;
+
+  span_test::mock_rng rng{0.5};
+
+  nuts::SpanW<double> span_from = span_test::dummy_span(1.8);
+  nuts::SpanW<double> span_to = span_test::dummy_span(0.3);
+  double new_logp = nuts::log_sum_exp(span_from.logp_, span_to.logp_);
+
+  // metropolis: 0.3 - 1.8 = -1.5 < -0.3, rejects
+  {
+    nuts::SpanW<double> span_old = span_from;
+    nuts::SpanW<double> span_new = span_to;
+    nuts::SpanW<double> combined =
+        nuts::combine<nuts::Update::Metropolis, direction>(
+            rng, std::move(span_old), std::move(span_new));
+
+    EXPECT_SELECTED_POINT_EQUAL(combined, span_from, new_logp);
+  }
+
+  // barker: 0.3 - lsexp(0.3, 1.8) = 0.3 - 2.00 = -2.00 < -0.3, rejects
+  {
+    nuts::SpanW<double> span_old = span_from;
+    nuts::SpanW<double> span_new = span_to;
+    nuts::SpanW<double> combined =
+        nuts::combine<nuts::Update::Barker, direction>(rng, std::move(span_old),
+                                                       std::move(span_new));
+
+    EXPECT_SELECTED_POINT_EQUAL(combined, span_from, new_logp);
+  }
+}
+
+TYPED_TEST(CombineSpansSymmetric, BarkerAndMetropolisBothAccept) {
+  // cases where the two update rules differ
```

Comment:
The update rules differ but they both accept?  I find this doc confusing given the name of the test.